### DISCUSSION TITLE
Issue: Increased CPU and Memory Overhead with Uvicorn Multi-Processing

### DISCUSSION THREAD

**Discussion by jahnavisana2812** [Kludex/uvicorn#2463] [Potential Issue]
We were previously using Gunicorn with Uvicorn workers for our application. Recently, we decided to shift to using Uvicorn directly, due to improvements in Uvicorn.

However, while Gunicorn used to spawn 5 sub-processes for handling requests, Uvicorn now creates 5 multiprocessing spawn processes using Python's multiprocessing library. This results in noticeable CPU and memory overhead, which we didn't experience with Gunicorn workers.

When running Uvicorn with a single worker, the overhead is not present, but increasing the worker count leads to the issue. I have attached screenshots of our monitoring data, which clearly show the difference in resource usage between configurations.

Is this behavior expected with Uvicorn, or is there a possible fix for reducing the overhead in future updates? Any guidance or suggestions would be appreciated.

Python version:3.11.9
Uvicorn version:0.30.6
With default worker=1
<img width="1419" alt="Screenshot 2024-09-20 at 4 20 18 PM" src="https://github.com/user-attachments/assets/d3151111-0c9d-45ba-8e20-0384e3ae2d5b">
with workers=5
<img width="1438" alt="Screenshot 2024-09-20 at 4 20 39 PM" src="https://github.com/user-attachments/assets/5fe3f56e-af88-4b6f-8fb0-66bcd56c74a3">

**Comment by nikhil-weamai** [comment]
same issue I m facing with higher memory usages

**Reply by theyashl** [reply]
You can check the associated issue thread https://github.com/encode/uvicorn/issues/2466 if it helps.
